aries_cloudagent/anoncreds/base.py

```diff
@@ -144,7 +144,7 @@ async def register_schema(
         self,
         profile: Profile,
         schema: AnonCredsSchema,
-        options: Optional[dict] = None,
+        options: dict = {},
```

Comment:
FYI, default empty dicts in args are 'dangerous' in python:
https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

Pylint warns about it, so ruff should probably pick up on this? @dbluhm (because you know about the ruff config)

Reply:
Really good point. Looks like this would be covered by rule [B006](https://docs.astral.sh/ruff/rules/mutable-argument-default/), which we don't currently have activated.

@jamshale it would be great to see a follow up PR to add checks for this rule and fix this argument.

Reply:
I'll have another PR for the publish revocations and I'll do that.

Reply:
I had no idea python did that. Seems really odd a default function argument wouldn't be scoped to the function.

Reply:
I have personal experience with this foot-gun lol. Debugging without the context of the "static" evaluation of the default arguments was a nightmare. The Ruff check will be good to help prevent future headaches.